src/Orchard/Mvc/MvcModule.cs

```diff
@@ -145,12 +170,32 @@ public class HttpResponsePlaceholder : HttpResponseBase {
         /// standin context for background tasks. 
         /// </summary>
         public class HttpRequestPlaceholder : HttpRequestBase {
+            private HttpContextBase _httpContext;
             private readonly Uri _uri;
 
-            public HttpRequestPlaceholder(Uri uri) {
+            public HttpRequestPlaceholder(HttpContextBase httpContext, Uri uri) {
+                _httpContext = httpContext;
                 _uri = uri;
             }
 
+            public override RequestContext RequestContext {
+                get {
+                    return new RequestContext(_httpContext, new RouteData());

```

Comment:
Add a local variable that is initialized on this call, and return it on the next calls.